homeassistant/components/switch/tahoma.py

```diff
@@ -0,0 +1,51 @@
+"""
+Support for Tahoma Switch - those are push buttons for garage door etc.
+
+Those buttons are implemented as switchs that are never on. They only
+receive the turn_on action, perform the relay click, and stay in OFF state
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.tahoma/
+"""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.components.tahoma import (
+    DOMAIN as TAHOMA_DOMAIN, TahomaDevice)
+
+DEPENDENCIES = ['tahoma']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Tahoma switchs."""
+    controller = hass.data[TAHOMA_DOMAIN]['controller']
+    devices = []
+    for switch in hass.data[TAHOMA_DOMAIN]['devices']['switch']:
+        devices.append(TahomaSwitch(switch, controller))
+    add_devices(devices, True)
+
+
+class TahomaSwitch(TahomaDevice, SwitchDevice):
+    """Representation a Tahoma Switch."""
+
+    @property
+    def device_class(self):
+        """Return the class of the device."""
+        if self.tahoma_device.type == 'rts:GarageDoor4TRTSComponent':
+            return 'garage'
+        return None
+
+    def turn_on(self, **kwargs):
```

Comment:
I don't think you need this. `switch.toggle` works just fine, right?

Reply:
I'll try to remove it to check that, but I think I tryed.
I think I couldn't have it displayed on GUI then, or that the GUI does not offer the "toggle" button. I might have miss something ?

Reply:
As expected:
  File "/mnt/d/git/home-assistant/homeassistant/helpers/entity.py", line 388, in turn_on
    raise NotImplementedError()

The switch exposed on the GUI uses on/off function. It would probably possible to call toggle directly, but in order to have a more user friendly interface, I think it is better to keep the "turn_on" function.